packages/date-picker/src/DatePicker/DatePicker.spec.tsx

```diff
@@ -283,25 +276,21 @@ describe("<DatePicker /> - Input format", () => {
   it("formats values when focus is on the input", async () => {
     render(<DatePickerWrapper selectedDay={new Date("2022-03-01")} />)
 
-    const input = screen.getByLabelText("Input label", { selector: "input" })
+    const input = screen.getByRole("combobox", { name: "Input label" })
     expect(input).toHaveValue("Mar 1, 2022")
 
-    await waitFor(() => {
-      input.focus()
-    })
+    userEvent.tab()
```

Comment:
If there is any change in the dom structure, this will fail. The idea was that because `focus` on the input could be triggered through click or keyboard navigation, calling `.focus` would cater for both.

Nonetheless, you're right that calling `.focus` probably isn't the best way to replicate user intention, so I'd say it would be better to click on the input to be more explicit. Or, if we are using `tab`, then to first assert that the input has the focus.

Reply:
The intent here is that “the user is real life will hit the tab key” but I understand if it looks a bit loose. But I believe that is the ‘contract’ that we are having with the user?

Reply:
One idea I had was to have one test that verifies that tabbing focuses the input we want, and then always just tab in other tests.